src/Controller/ControllerProject.php

```diff
@@ -215,6 +219,13 @@ public function __construct($entityManager, $user)
                     $this->entityManager->persist($projectDuplicated);
                     $this->entityManager->flush();
 
+                    // save Exercise and related unit tests
+                    $success = $this->assignRelatedExercicesAndTestsToStudent($project,$projectDuplicated);
+                   
+                    if(!$success){
+                        return array('error'=> "ExercisesAndUnitTestsNotSavedProperly");
```

Comment:
Why not:  
```
if (!$this->assignRelatedExercicesAndTestsToStudent($project,$projectDuplicated)) 
return array('error'=> "ExercisesAndUnitTestsNotSavedProperly");
```

You named the method very well. To the point you made it natural to read `if (!$this->assignRelatedExercicesAndTestsToStudent($project,$projectDuplicated)) ` ; I read this as : "If I could not assign a related exercise and test to student". Bravo for the method name 👏